### PR TITLE
Refactor tests and add "add facility" test

### PR DESCRIPTION
**Pull request by schra**

### CODE REVIEW

File: tests/framework.bash [hunk 1]
```diff
@@ -22,21 +19,58 @@ containers-stop() {
...
+now() {
+  date +%s%3N
+}
+
+# repeat a command until it succeeds, or until timeout
+await() {
+  local timeout=$1; shift
+  local command=("$@")
+
+  local now_
```
Comment (aticu) [inline tests/framework.bash]: I think something like `startTime` would be better than `now_` here.

File: tests/framework.bash [hunk 2]
```diff
@@ -22,21 +19,58 @@ containers-stop() {
...
+  done
+  return 1
+}
+
+# Wait until the HTTP server of the Docker container is reachable. We give up
+# after a certain amount of waiting time and assume that the HTTP server failed
+# to start.
+await-http() {
+  # the requested path has to be a path the server serves so that the successful startup test doesn't fail
+  await 5000 curl -sS --max-time 1 --connect-timeout 1 "http://$TONARI_IP:8000/facilities/by-radius/1/1/1"
```
Comment (aticu) [inline tests/framework.bash]: I think I'd prefer the request `/facilities/by-tile/0/0/0` here. The current request requests facilities at an arbitrary location, while this one would request all facilities.

Or we could comment that it is just an arbitrarily chosen request.

If we do neither of these things I fear that a reader of this code may try to find the significance of this request.

File: tests/sigint-stops-docker-container.bats [hunk 3]
```diff
@@ -4,20 +4,16 @@ load framework
...
-    echo Tonari is running > /dev/null
-    return 1
-  fi
-
-  docker kill --signal=SIGINT $TONARI > /dev/null
-  sleep 4
-  if container-running $TONARI; then
-    log Tonari failed to be killed
+  # Did the container start?
+  if ! container-running "$TONARI"; then
```
Comment (aticu) [inline tests/sigint-stops-docker-container.bats]: I might be mistaken here, but isn't this what's already being checked by `await-http` in `setup()`?

**Review by aticu — CHANGES_REQUESTED** [review]

File: tests/framework.bash (hunk 2 above)
Comment (schra) [reply tests/framework.bash]: Indeed this makes more sense.

File: tests/sigint-stops-docker-container.bats (hunk 3 above)
Comment (schra) [reply tests/sigint-stops-docker-container.bats]: Oh yeah it is.

**Review by aticu — APPROVED** [review]

**Pull request merged**